### PR TITLE
Refactor User Profile unit tests  AB#10077

### PR DESCRIPTION
**Pull request by pagustinqt20**
# Refactor User Profile unit tests  [AB#10077](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/10077)

-   [x] Enhancement
-   [ ] Bug
-   [ ] Documentation

## Description

Refactor the userprofileservice unit tests. Introduce new classes to have a reusable tests. 

If you are reviewing this PR, please adhere to the guidelines as [documented](https://github.com/bcgov/healthgateway/wiki/prguidance).

## Testing

-   [x] Unit Tests Updated
-   [ ] Functional Tests Updated
-   [ ] Not Required

### Steps to Reproduce

If this is a bug, please provide details on how to reproduce the code.

### UI Changes

NO

### Browsers Tested

-   [ ] Chrome
-   [ ] Safari
-   [ ] Edge
-   [ ] Firefox

Provide an explanation for any test(s) not completed.

## Notes/Additional Comments

Any additional notes or comments that may be relevant. Include any specialized deployment steps here.

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

-   Fulfills Work Item Requirements

    -   The changes meet/implement story's acceptance criteria. Fixes identified problems if bug.
    -   Changes not directly related to the task requirements need to be documented on the PR.

-   Compilation / Tests

    -   The changes do not break compilation and/or unit or functional tests; unless other item addresses them specifically.

-   Logic Problems / Functional Behaviour

    -   The changes work as intended and do not introduce problems.

-   Performance

    -   The changes do not introduce obvious performance issues.

-   Documented Standards

    -   The changes adhere to the team's documented [Coding Standards](https://github.com/bcgov/healthgateway/wiki/standards).

-   Readability / Maintainability
    -   The changes can be easily understood/read and allow for future enhancements without major refactoring. Readability is preferred over "clever code".
    -   Reasoning for changes needs to be clearly articulated. Disagreements should be arbitrated by a third developer.

### CODE REVIEW

**Review by sslaws — DISMISSED** [review]

**Review by BrianMaki — DISMISSED** [review]

File: Apps/WebClient/test/unit/Services.Test/Mock/UserPreferenceDelegateMock.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        {
+            this.Setup(s => s.GetUserPreferences(hdid)).Returns(readResult);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="UserPreferenceDelegateMock"/> class.
+        /// </summary>
+        /// <param name="readResult">read result.</param>
+        /// <param name="action">action.</param>
+        public UserPreferenceDelegateMock(DBResult<UserPreference> readResult, string action)
```
Comment (MikeLyttle) [inline Apps/WebClient/test/unit/Services.Test/Mock/UserPreferenceDelegateMock.cs:46]: I think `action` should be an enum for stronger typing.

File: Apps/WebClient/test/unit/Services.Test/UserProfileServiceTests.cs [hunk 2]
```diff
@@ -206,595 +216,343 @@ public async Task ShouldQueueNotificationUpdate()
...
-
-            return new Tuple<RequestResult<UserProfileModel>, UserProfileModel>(actualResult, expected);
+            // Assert
+            Assert.Equal(ResultType.Success, actualResult.ResultStatus);
+            Assert.Null(actualResult.ResourcePayload?.ClosedDateTime);
         }
 
-        private Tuple<RequestResult<UserProfileModel>, UserProfileModel> ExecuteGetUserProfile(DBStatusCode dbResultStatus, DateTime lastLoginDateTime)
+        /// <summary>
+        /// RecoverUserProfile - Alraedy active happy Path.
```
Comment (MikeLyttle) [inline Apps/WebClient/test/unit/Services.Test/UserProfileServiceTests.cs:524]: Typo: Alraedy should be Already.

**Review by MikeLyttle — CHANGES_REQUESTED** [review]

File: Apps/WebClient/test/unit/Services.Test/Mock/UserPreferenceDelegateMock.cs (hunk 1 above)
Comment (pagustinqt20) [reply Apps/WebClient/test/unit/Services.Test/Mock/UserPreferenceDelegateMock.cs:46]: @MikeLyttle  I think declaring static const, is enough in this scenario?

Comment (pagustinqt20) [reply Apps/WebClient/test/unit/Services.Test/Mock/UserPreferenceDelegateMock.cs:47]: @MikeLyttle  I used const to avoid any casting

Comment (MikeLyttle) [reply Apps/WebClient/test/unit/Services.Test/Mock/UserPreferenceDelegateMock.cs:47]: I think we normally use enums, but this should be fine.

**Review by MikeLyttle — APPROVED** [review]

**Review by BrianMaki — APPROVED** [review]

**Pull request merged**